Realm/RLMSyncPermission.h

```diff
@@ -16,7 +16,250 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
-#import <Foundation/Foundation.h>
+#import <Realm/RLMObject.h>
+
+@protocol RLMPermission, RLMPermissionUser;
+@class RLMPermission, RLMPermissionUser, RLMPermissionRole,
+       RLMArray<RLMObjectType>, RLMLinkingObjects<RLMObjectType: RLMObject *>;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ A permission which can be applied to a Realm, Class, or specific Object.
+
+ Permissions are applied by adding the permission to the RLMRealmPermission singleton
+ object, the RLMClassPermission object for the desired class, or to a user-defined
+ RLMArray<RLMPermission> property on a specific Object instance. The meaning of each of
+ the properties of RLMPermission depend on what the permission is applied to, and so are
+ left undocumented here. See `RLMRealmPrivileges`, `RLMClassPrivileges`, and
+ `RLMObjectPrivileges` for details about what each of the properties mean when applied to
+ that type.
+ */
+@interface RLMPermission : RLMObject
+/// The Role which this Permission applies to. All users within the Role are
+/// granted the permissions specified by the fields below any
+/// objects/classes/realms which use this Permission.
+@property (nonatomic) RLMPermissionRole *role;
+
+@property (nonatomic) bool canRead;
```

Comment:
Should we be using `BOOL` rather than `bool` since this is Objective-C? Is `bool` even available in Objective-C by default without including `stdbool.h`?… I guess it must be, since this all appears to compile.

Reply:
Foundation.h pulls in stdbool.h even on platforms where BOOL isn't a typedef for bool.